### PR TITLE
fix #36 by adding a new translation table ...

### LINKED ISSUE #36: Add support for selenoproteins
bioutils currently does not support selenoproteins. It would be great to be able to add it to the codon translation table ([technically it is already there](https://github.com/biocommons/bioutils/blob/ef5310aa061caf92170ba36c5d62a34b51f79ccc/src/bioutils/sequences.py#L35), but maps to "") .  We prob want to have a different translation table that maps the [stop codon](https://github.com/biocommons/bioutils/blob/ef5310aa061caf92170ba36c5d62a34b51f79ccc/src/bioutils/sequences.py#L97) to SEC / U. Also add support for the alternate translation table as an option to [translate_cds](https://github.com/biocommons/bioutils/blob/ef5310aa061caf92170ba36c5d62a34b51f79ccc/src/bioutils/sequences.py#L493).

Finally I don't think that sequences.py has any test coverage. Add some bare bones unit tests for the newly added features.

### PR DESCRIPTION
**Pull request by andreasprlic**
that adds support for SEC/ to the translation table

### CODE REVIEW

**Comment by andreasprlic** [comment]
looks like the DeepSource plugin ^^^ is complaining about a ton of unrelated details in the repo.

**Comment by reece** [comment]
@andreasprlic : I'll review tomorrow.

**Comment by andreasprlic** [comment]
@reece In order to lighten the load on your shoulders, what do you think of granting more people admin access to all of biocommons. I was specifically thinking of including the GA4GH VRS folks @ahwagner and @larrybabb ?

File: src/bioutils/sequences.py [hunk 1]
```diff
@@ -490,10 +493,11 @@ def replace_u_to_t(seq):
...
 
-def translate_cds(seq, full_codons=True, ter_symbol="*"):
+def translate_cds(seq, full_codons=True, ter_symbol="*", translation_table=dna_to_aa1_lut):
     """Translates a DNA or RNA sequence into a single-letter amino acid sequence.
     
-    Uses the NCBI standard translation table.
+    By default uses the NCBI standard translation table. To enable translation for selenoproteins,
+    the dna_to_aa1_sec table can get used.
 
     Args:
```
Comment (gomoto) [inline src/bioutils/sequences.py:522]: Can you add an entry for `translation_table` to the `Args` section?

File: tests/test_sequences.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    assert translate_cds("GAS") =='X'
+    assert translate_cds("CUN") == 'L'
+    with pytest.raises(ValueError):
+        translate_cds("AUGCGQ")
+
+def test_translate_selenoproteins():
+    """ unit test for sec codon"""
+    assert translate_cds("AUGTGATAA") == 'M**'
+    assert translate_cds("AUGTGATAA", translation_table=dna_to_aa1_lut) == 'M**'
+    assert translate_cds("AUGTGATAA", translation_table=dna_to_aa1_sec) == 'MU*'
```
Comment (gomoto) [inline tests/test_sequences.py]: For completeness, what about the case where both `full_codons` and `translation_table` are set to non-default values?

**Comment by gomoto** [comment]
A couple of the DeepSource issues are related to this PR:
- https://deepsource.io/gh/biocommons/bioutils/run/a8aaccdf-d194-4c7a-ae72-f328bcf4584b/python/FLK-E225
- https://deepsource.io/gh/biocommons/bioutils/run/a8aaccdf-d194-4c7a-ae72-f328bcf4584b/python/FLK-D210

But most are not. Is there a way to trigger this check on main to confirm that this PR didn't introduce all of the other issues?

**Comment by andreasprlic** [comment]
I suspect and admin could trigger deepsource on the main branch, but I don't have admin privileges.

**Review by gomoto — APPROVED** [review]
The formatting change created a lot of noise (probably would have been better as its own PR), but the core change looks good. 

@reece What do you think?

**Comment by reece** [comment]
So sorry for the delay, @andreasprlic . Looks good to me! I'll merge, and plan to address the DeepSource issues later.

**Pull request merged**